app/models/activity_logger.rb

```diff
@@ -70,12 +83,82 @@ def close(duration: true)
     record('info', "Finished at #{finish_time}.")
     record('info',
       "Duration: #{(finish_time - @start_time).round(2)} seconds.\n") if duration
-    @log.close
+    @log.close if @log.respond_to?(:close) && !@is_system_outstream
     logged_to if @show
   end
 
   def logged_to
     puts @facility_and_action + '[info] ' + "Information was logged to: #{@filename}"
   end
 
+
+  # Return a stream to use that we have verified is writeable.
+  #
+  # Given a full filename path, ensure that we can write to the directory.
+  # If not, use $stdout instead.
+  # If the directory for _filename_ does not exist, try to create it.
+  # If we cannot create the directory, then use $stdout for the output.
+  # If the directory is not writeable, then use $stdout for the output.
+  #
+  # @param unverified_filename [String] - the full path of the unverified filename; we
+  #   try to verify that the directory for it exists and is writeable
+  #
+  # @return the full filename + path to use for output.
+  #      This might be changed to $stdout if we could not use or create the directory
+  #      for filename
+  def self.verified_output_stream(unverified_filename)
+
+    @is_system_outstream = true
+
+    # always log to stdout no matter what unverified_filename is
+    if ENV.has_key?('RAILS_LOG_TO_STDOUT')
+      return $stdout
+    end
+
+    # gating condition:
+    return unverified_filename if is_system_stream?(unverified_filename)
+
+    # if we're here, we didn't explicitly use a system output stream
+    @is_system_outstream = false
+
+    verified_output = $stdout # fallback to this unless we can verify the directory
+
+    unverified_dir = File.dirname unverified_filename
+
+    begin
+      verified_output = unverified_filename if dir_verfied?(unverified_dir)
+    rescue => _err
+      # If we can't create or write to the directory, then this
+      # directory is not verified.  Use the fallback output ($stdout)
+    end
+
+    verified_output
+  end
+
+
+  def self.is_system_stream?(stream)
+    (stream == $stdout || stream == $stderr || stream.class.name == 'IO')
+    # sometimes a stdout will not have the same address as $stdout depending on
+    # how it was created. Hence we also test the class name
+  end
+
+
+  def self.dir_verfied?(unverified_dir)
```

Comment:
`dir_verified?`

Reply:
LOL - no, I was just making a comment on the spelling of `verified`.  Sorry to be so cryptic!